docs/advanced_usage.rst

```diff
@@ -0,0 +1,333 @@
+Advanced Usage
```

Comment:
I mostly, aside from fixing typos and cleaning up the `ddtrace-run` docs just moved these docs from `index.rst`. I feel like there is a lot here that has to be updated or rewritten.